### PR TITLE
[WNMGDS-1877] filter chip, pagination, & spinner test updates

### PR DESCRIPTION
**Pull request by scheul93**
## Summary

- Converting FilterChip, Pagination & Spinner tests to React Testing Library

## How to test

1. Instructions on how to test the changes in this PR.

### If this is a change to code:

- [X] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[ticket] title`
- [X] Ran `yarn type-check && yarn lint && yarn test` with no errors

### CODE REVIEW

**Review by forrestbaer — APPROVED** [review]

File: packages/design-system/src/components/FilterChip/FilterChip.test.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    const props = { ...defaultProps, ...customProps };
+    return render(<FilterChip {...props} />);
+  };
+
+  it('should include children as label', () => {
+    renderFilterChip();
+    const chipEl = screen.getByRole('button');
+    expect(chipEl.textContent).toEqual('Foo. Remove Foo filter .');
+
+    expect(chipEl.innerHTML).toMatchInlineSnapshot(
```
Comment (pwolfert) [inline packages/design-system/src/components/FilterChip/FilterChip.test.tsx:21]: What's the benefit of this being inline vs in a snapshot file?

Comment (scheul93) [reply packages/design-system/src/components/FilterChip/FilterChip.test.tsx:21]: Not sure if there's a benefit. Is it harder to read if it's inline?

Comment (pwolfert) [reply packages/design-system/src/components/FilterChip/FilterChip.test.tsx:21]: All our other snapshots are in separate files, so I was wondering if this was a special case for a hidden reason. I do think it's a little easier, but I'm mainly interested in consistency

File: packages/design-system/src/components/FilterChip/FilterChip.test.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      renderFilterChip({ onDelete: onDelete });
+
+      const chipEl = screen.getByRole('button');
+      expect(chipEl).toBeDefined();
+
+      userEvent.click(chipEl);
+      expect(onDelete).toHaveBeenCalled();
+    });
+
+    it('should call onDelete when certain keyboard keys are pressed', () => {
```
Comment (pwolfert) [inline packages/design-system/src/components/FilterChip/FilterChip.test.tsx:42]: 100% test coverage, and they're concise and clear!

**Review by pwolfert — APPROVED** [review]
🎉

File: packages/design-system/src/components/Pagination/__snapshots__/Page.test.tsx.snap [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
-  <Button
-    aria-label="page 1"
-    href="/#1"
-    onClick={[MockFunction]}
-    variation="ghost"
-  >
-    1
-  </Button>
-</li>
+<div>
```
Comment (zarahzachz) [inline packages/design-system/src/components/Pagination/__snapshots__/Page.test.tsx.snap:4]: Something I noticed doing my tests was RTL adds an extra wrapping div to components. I don't know if this is an issue or not with our snapshots, but it's worth calling out that this wrapping div may not be part of the actual component we're testing.

Comment (scheul93) [reply packages/design-system/src/components/Pagination/__snapshots__/Page.test.tsx.snap]: Good call. I updated to target the element that `Page` actually renders. This makes it more consistent with your accordion tests.

Comment (pwolfert) [reply packages/design-system/src/components/Pagination/__snapshots__/Page.test.tsx.snap]: @zarahzachz, if you use `container` to take the snapshot, that container defaults to a `div` and will therefore wrap your snapshot in a div. If you use `asFragment()`, which we do in some of our other tests, the wrapping element in the snapshots will be a `<DocumentFragment>`. I'm a fan of trying to target the specific element you want to snapshot in general, as Anna suggested; it makes for smaller snapshots with less noise.

**Pull request merged**